packages/react/src/BubbleMenu/BubbleMenuFactory.tsx

```diff
@@ -0,0 +1,52 @@
+import { Editor, isNodeSelection, posToDOMRect } from "@tiptap/core";
+import { createRoot } from "react-dom/client";
+import { MantineProvider } from "@mantine/core";
+import { BlockNoteTheme } from "../../../core/src/BlockNoteTheme";
+import { BubbleMenu } from "./components/BubbleMenu";
+import tippy from "tippy.js";
+import rootStyles from "../../../core/src/root.module.css";
+
+export const BubbleMenuFactory = (editor: Editor) => {
```

Comment:
should this just return an element? Or an object with methods like: updatePosition, show, close, etc? Or is this still WIP / we'll figure this out later?

Reply:
I was also thinking about that, but I'm still not sure what the best approach regarding menu factory parameters and return types is.

That's something I experimented with in `MenuFunctions`, which would get passed to a factory and provide all the functions necessary for someone's custom UI to figure out on its own what & when to update. This approach IMO gives users much more flexibility with the UI.

The main concern I have is figuring out all the functions users might want when making their UI. For example, if someone wanted to display the names of editor's extensions for some reason, it doesn't really make sense to add a function which gets the names of the editor's extensions to `MenuFunctions` just for this niche use case. If we want to allow that level of customization, we'd be better off passing the whole editor instance to the custom UI like we do now.

It's a similar situation with that the factory should return. If the user wants to make a static menu, for example, it doesn't make sense for BlockNote to decide when the menu should show/hide.

So really, both the factory parameters and return types depend on how much flexibility we want to give the user. On one hand we can pass the editor instance as an argument and expect to receive a HTML element, letting (but also forcing) the user to decide how the menu should behave. On the other, we pass nothing and expect to receive not only a HTML element, but also show/hide/update functions for it, leaving it entirely up to the `BubbleMenuPlugin` to decide how the menu should behave.

Reply:
I think this is something we should discuss once the PR is closer to being finished, but it's also possibly something we might want to discuss with the Fermat guys.

Reply:
Let's now focus on the scenario where someone wants to build the existing functionality, but without React. That way we have a "locked" set of things that should be necessary. If then later we notice a lot more different customizations are needed, we can implement a more versatile solution like "editor" integration etc.

imo the show / hide / update functions are necessary in this case for clients to show the menu when text is highlighted, and removed when the focus changes. You're right that this might be different if creating a menu that is always visible, I'm wondering if we can think of an API that works for both (maybe "onSelectionChanged" or onBlur or sth, not sure what makes sense exactly

Reply:
In that case, can we just use a regular class to make the object:
```
{
    element: HTMLElement,
    show: () => void,
    hide: () => void,
    update: (params: BubbleMenuUpdateParams) => void
}
```
instead of a factory? Since we don't need to redefine `show`, `hide`, and `update`, we just potentially need to create, destroy, or update `element`.